File: playpen/swigtest/libsrc/Bar.cc
// -*- C++ -*-
//
// ======================================================================
//
//                           Brad T. Aagaard
//                        U.S. Geological Survey
//
// {LicenseText}
//
// ======================================================================
//


#include "Bar.hh"

#include <iostream>
#include <stdexcept>

geometry::Bar::Bar(void) :
  _length(0),
  _width(0)
{}
geometry::Bar::~Bar(void)
{}

void
geometry::Bar::length(const double value) {
  if (value <= 0)
    throw std::runtime_error("Length must be positive.");
  _length = value;
}
double
geometry::Bar::length(void) const {
  return _length;
}

void
geometry::Bar::width(const double value) {
  if (value <= 0)
    throw std::runtime_error("Width must be positive.");
  _width = value;
}
double
geometry::Bar::width(void) const {
  return _width;
}

void
geometry::Bar::view(void) const {
  std::cout << "Bar:\n"
	    << "  length: " << _length << "\n"
	    << "  width: " << _width << "\n"
	    << "  color: " << _color << "\n"
	    << "  id: " << _id << "\n"
	    << std::endl;
}


// End of file

File: playpen/swigtest/python/Bar.py
#!/usr/bin/env python
#
# ----------------------------------------------------------------------
#
#                           Brad T. Aagaard
#                        U.S. Geological Survey
#
# <LicenseText>
#
# ----------------------------------------------------------------------
#

from Shape import Shape
from geometry import Bar as ModuleBar

# Bar class
class Bar(Shape, ModuleBar):

  # INVENTORY //////////////////////////////////////////////////////////

  class Inventory(Shape.Inventory):
    """Python object for managing Bar facilities and properties."""

    import pyre.inventory

    length = pyre.inventory.float("length", default=1.0)
    length.meta['tip'] = "Length of bar."

    width = pyre.inventory.float("width", default=1.0)
    width.meta['tip'] = "Width of bar."


  # PUBLIC METHODS /////////////////////////////////////////////////////

  def __init__(self, name="bar"):
    """
    Constructor.
    """
    Shape.__init__(self, name)
    return


  # PRIVATE METHODS ////////////////////////////////////////////////////

  def _configure(self):
    """
    Setup members using inventory.
    """
    Shape._configure(self)

    self.length(self.inventory.length)
    self.width(self.inventory.width)
    return


  def _createModuleObj(self):
    ModuleBar.__init__(self)
    return


# FACTORIES ////////////////////////////////////////////////////////////

def shape():
  return Bar()


# End of file 
